lib/storage/src/content_manager/toc/collection_meta_ops.rs

```diff
@@ -199,13 +199,25 @@ impl TableOfContent {
             drop(removed);
 
             // Move collection to ".deleted" folder to prevent accidental reuse
-            let uuid = Uuid::new_v4().to_string();
+            // the original collection path will be moved atomically within this
+            // directory.
             let removed_collections_path =
                 Path::new(&self.storage_config.storage_path).join(".deleted");
             tokio::fs::create_dir_all(&removed_collections_path).await?;
-            let deleted_path = removed_collections_path
-                .join(collection_name)
-                .with_extension(uuid);
+
+            let temp_file_binding = Builder::new()
+                // Limit the file name to be on a lower side to avoid running into too-long
+                // file names.
+                // Even if the chosen randomness factor poses chances of collision, the library
+                // prevents creation of duplicate files within the chosen directory.
+                .rand_bytes(8)
+                .tempfile_in(removed_collections_path)?;
```

Comment:
I think you need tempdir instead:

```suggestion
                .tempdir_in(removed_collections_path)?;
```

As far as I understand the `tokio::fs::rename` will simply replace it if the directory is still empty.

Reply:
> I think you need tempdir instead:

both ```tempdir``` and ```tempfile``` crates are similar.  In qdrant, we use tempfile, here's a [github search](https://github.com/search?q=repo%3Aqdrant%2Fqdrant+tempfile+path%3A*.toml&type=code); so i didn't feel like adding a new dependency.

Reply:
> oh i think you meant, using tempdir_in instead of tempfile_in using the tempfile crate, let me check on it.

Correct :smile: 

Just because it better matches with the end result, having a directory there (not a file).

Reply:
You are right.  I misjudged that the ```path``` that is being renamed to be a file, its actually a directory.

Reply:
Thanks for the patch!